staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -174,28 +174,39 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []
 					hasSubRequests: true,
 				}
 				requestKey.subRequest = true
+				minDevicesPerRequest := 0
 				for i, subReq := range request.FirstAvailableOf {
 					reqData, err := alloc.validateDeviceRequest(requestAccessor{subRequest: &subReq}, requestKey, pools)
 					if err != nil {
 						return nil, err
 					}
 					requestKey.subRequestIndex = i
 					alloc.requestData[requestKey] = reqData
+					if reqData.numDevices > minDevicesPerRequest {
```

Comment:
I don't think this is correct, as it computes the max number of devices per request. But I updated it when I incorporated the changes in my PR and also added a test.

Reply:
The intent was to determine the minimum number of devices which must be allocated for the request and ultimately for the claim. If that number is too large, there is no way how the claim can be allocated, so we don't even need to try.